src/Storefront/Framework/Routing/RequestTransformer.php

```diff
@@ -314,6 +314,13 @@ private function resolveSeoUrl(Request $request, string $baseUrl, string $langua
             $seoPathInfo = mb_substr($seoPathInfo, mb_strlen($baseUrl));
         }
 
+        // Include query string in resolving so SEO URLs stored with query parameters
+        // (e.g., "awesome-product?test=123") are matched exactly when present.
```

Comment:
but seo url `awesome-product` is no longer matched when the query string exists, which should not be the case

So IMO you should check 

WHERE seo_path_info = $pathInfo OR seo_path_info = $pathInfoWithQueryString